packages/server-common/src/utils/sync-google-event.ts

```diff
@@ -79,16 +80,34 @@ export const syncEventFactory =
         return eventsController.update(existingEvent.id, eventToUpdate);
       }
 
-      const eventToCreate = {
-        ...newEvent,
-        hidden: newEvent.status === 'Cancelled',
-        tags: [],
-      };
+      const hidden = newEvent.status === 'Cancelled';
+      if (isCRNEventController(eventsController)) {
+        const crnEventToCreate = {
+          ...newEvent,
+          hidden,
+          tags: [],
```

Comment:
so you say we can only add `tags` when it's a CRN event but if that was the case we would never be able to add to the input object for GP2 - can you make sure gp2 types reflect it?

Reply:
(I've left the same message on Slack but I believe it's better to have it here as well)

To be honest I don’t know why typescript didn’t catch that tags can’t be added if the controller can have the type gp2.EventController  🤔 

The types are already correct, the CRN controller with type EventController accepts this EventCreateDataObject  (with tags) https://github.com/yldio/asap-hub/blob/master/packages/model/src/event.ts#L82:L97 as the event type in create while the GP2 controller accepts this EventCreateDataObject (without tags) https://github.com/yldio/asap-hub/blob/master/packages/model/src/gp2/event.ts#L44:L59 as the event type

The only way I was able to make TS display an error was with this type guard

Reply:
I've spent some time trying to understand why it does not give an error the way it is on master now. I wrote this example in my VSCode

```
interface CreatePropsDog {
  age: number;
  name: string;
  owner: string;
  birthdate: string;
  breed: string;
}
interface Dog {
  create: (pet: CreatePropsDog) => string;
}

interface CreatePropsCat {
  age: number;
  name: string;
  owner: string;
  birthdate: string;
  color: string;
}

interface Cat {
  create: (pet: CreatePropsCat) => string;
}

let pet: Dog | Cat = {
  create: () => 'hey',
};

pet.create({
  age: 12,
  name: 'Pet',
  owner: 'Owner',
  birthdate: '28/01/2021',
  breed: 'Chihuahua',
  color: 'black',
});
```

And it seemed odd to me that I could pass `breed` only present in DogProps and `color` only present in CatProps without any TS errors and it turns out when I hover `create` it gives me this

<img width="755" alt="Screenshot 2023-09-15 at 14 05 31" src="https://github.com/yldio/asap-hub/assets/16595804/bd73ded6-8c87-4d22-8f80-8a0557ad5a81">

It seems Typescript makes a intersection of arguments when there's a union type https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-3.html#improved-behavior-for-calling-union-types

That's why it doesn't give an error the way it is now on master when we have `tags: []` in the create payload, because even though GP2 events don't have this field CRN events do.

Reply:
yeah I was obviously wrong when I said it - TypeScript will always accept an object with tags property because of reasons you said (intersection when using unions). There is no excess property checking either. So yeah we would need to find a way to somehow link this with a passed generic and enforce type checking when the factory is invoked (which is the either crn or gp2 handler at which point we know exactly which input type we are dealing with) but lets leave it for now. This works so i am approving